neqo-transport/src/qlog.rs

```diff
@@ -234,18 +226,31 @@ pub fn packet_sent(
                 }
             }
 
-            Some(EventData::PacketSent(PacketSent {
-                header,
-                frames: Some(frames),
-                is_coalesced: None,
-                retry_token: None,
-                stateless_reset_token: None,
-                supported_versions: None,
-                raw: Some(raw),
-                datagram_id: None,
-                send_at_time: None,
-                trigger: None,
-            }))
+            match meta.direction() {
+                Direction::Tx => Some(EventData::PacketSent(PacketSent {
+                    header: meta.into(),
+                    frames: Some(frames),
+                    is_coalesced: None,
+                    retry_token: None,
+                    stateless_reset_token: None,
+                    supported_versions: None,
+                    raw: Some(raw),
+                    datagram_id: None,
+                    send_at_time: None,
+                    trigger: None,
+                })),
+                Direction::Rx => Some(EventData::PacketReceived(PacketReceived {
```

Comment:
Does `PacketReceived` implement `Default`?  Because it seems like you could replace a bunch of this code with `..PacketReceived::default()`.

Reply:
I had the same thought, but it doesn't. Maybe I'll do a PR against qlog...

Reply:
https://github.com/cloudflare/quiche/pull/1931